app/src/main/java/org/mozilla/fenix/settings/address/ext/Address.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address.ext
+
+import mozilla.components.concept.storage.Address
+
+/**
+ * Generate a label from a given [Address]. A well-formed [Address] will have
+ * a full name in the label, otherwise it will find the highest priority [Address] property
+ * to use, as based on the desktop code found in: FormAutofillUtils::getAddressLabel
+ */
+fun Address.toLabel(): String = getName().ifEmpty { iterateFields("") }
+
+/**
+ * Generate a description from a given [Address]. A well-formed address will have a street address
+ * as a description. An address with missing data will use the next available property after [label]
+ * based on the priorities found in the desktop code in: FormAutofillUtils::getAddressLabel
+ */
+fun Address.toDescription(label: String): String = iterateFields(label)
+
+private fun Address.getName(): String {
+    val middleName = if (additionalName.isEmpty()) "" else "$additionalName "
+    return if (givenName.isNotEmpty()) "$givenName $middleName$familyName" else familyName
+}
+
+private fun Address.iterateFields(previous: String): String =
+    streetAddress.joinStreetAddress()
```

Comment:
`[streetAddress, addressLevel3, addressLevel2, organization, addressLevel1, ..].map(field -> field.isNotEmpty()).join(", ")` 

I think this will also simplify your current approach

Reply:
I am okay with not implementing the addressField override for now, but I do want us to investigate to see if we should ever be concern about it to have our bases covered. This does require some digging into individual bugzilla issues, code, tests, whatever you can find. I don't believe the original authors are at Mozilla anymore, but it will help if we did our due diligence based on anything we can find. If we did discover we needed this, I would file a separate issue to ensure we have that on file. 

> My implementation also differs in that it returns the 2 parts (label and description) separately since that's what the caller anticipates. Do you think it makes more sense to return it as a single string and allow the caller to split it?

Desktop returns their label all in one line compare to our desired mobile design. So, it's totally okay for us to separate this. Based on my simple test of adding my own address and looking at https://searchfox.org/mozilla-central/source/toolkit/components/formautofill/FormAutofillUtils.jsm#323, I have to imagine they are using the addressField override for different regions since I only see my `<Full Name>, <One line Street Address>` (no country, state, phone, email that you can see in the linked code).

Reply:
There's a lot of things to parse through here, but as a way forward we can keep to adding all the fields or we could simplify this to displaying the `<One line Street Address>` and follow up based on the investigation of how desktop is displaying the address label for US/CA.

Reply:
> I have to imagine they are using the addressField override for different regions since I only see my <Full Name>, <One line Street Address> (no country, state, phone, email that you can see in the linked code).

I believe that's because of [this line](https://searchfox.org/mozilla-central/source/toolkit/components/formautofill/FormAutofillUtils.jsm#356). It seems to use the first two, highest priority fields available. You can see that in action if you enter addresses without a name or one line address:

![image](https://user-images.githubusercontent.com/10427470/166342139-e8d47a99-aacf-4bd3-8b2b-0a8ace84b5f4.png)

This is what I've tried to emulate on the mobile side, where `label` and `description` are the elements on desktop that are comma separated.

> This does require some digging into individual bugzilla issues, code, tests, whatever you can find. I don't believe the original authors are at Mozilla anymore, but it will help if we did our due diligence based on anything we can find.

This feels like it could be a pretty large undertaking. I think we should consider filing a separate spike if it's important enough to do that level of discovery for. Searching for usages of [getAddressLabel](https://searchfox.org/mozilla-central/search?q=getAddressLabel&path=) shows that the `addressFields` parameter defining the override isn't currently used. Would it maybe be easier to just track down a desktop PM and check if it's a feature they intend to continue supporting?

Reply:
> I believe that's because of [this line](https://searchfox.org/mozilla-central/source/toolkit/components/formautofill/FormAutofillUtils.jsm#356). It seems to use the first two, highest priority fields available. You can see that in action if you enter addresses without a name or one line address:
> 

Ok this line does make sense based on what we are seeing. I am kinda satisfied with that wrt to the addressFields override, and it doesn't look like it is used based on my quick search. So, I think we can move on from this topic.